test/fuzz/CMakeLists.txt

```diff
@@ -51,7 +51,9 @@ target_link_libraries(fuzztest-base
     ${PROJECT_NAME}::headers
     ${PROJECT_NAME}::common
     -fsanitize=fuzzer)
-target_compile_options(fuzztest-base PRIVATE -g -fsanitize=fuzzer)
+# When built with -g and -flto (which is required by some hardening flags), this causes a segfault in (upstream)
+# LLVM 14-15 while linking when CMAKE_BUILD_TYPE is Release
```

Comment:
should we have an issue for this?

Reply:
I don't think so. It feels like an upstream compiler bug, and we don't need to build with `-g` here anyway.

Reply:
Maybe an upstream issue is appropriate then?